src/modules/lifecycle.ts

```diff
@@ -74,12 +65,18 @@ class LifecycleApi extends BaseApi {
      *     myClientApp.lifecycle.bootstrapped();
      *   }, 500);
      * });
+     * ```
+     * 
+     * @param listener - The function to call when PureCloud is ready for the app to
+     * perform post-load initialization work.  This function will be passed the lifecycle event and
+     * does not augment the this context.
+     * @param once - If the listener should only be invoked once or repeatedly; true by default.
      *
      * @see [Advanced Application Concepts - Lifecycle Events](/api/client-apps/advanced.html#lifecycle_events)
      *
      * @since 1.0.0
      */
-    addBootstrapListener(listener, once = true) {
+    addBootstrapListener(listener: (event: SDKMessagePayload) => void, once = true) {
```

Comment:
Similar question here.  Would it be better to explicitly specify these using the MessageListener type?

Reply:
So yeah, this is probably my least favorite sacrifice that I had to make for typedoc to play nicely. Ideally, we would just import the MessageListener type and use that in all of these public method signatures. When taking this approach, typedoc's generated output only gives you `addBootstrapListener(listener: MessageListener, once: boolean): void` and it won't further expand into what the `MessageListener` type actually is (unless you give it an entire set of pages for interfaces). If you just give the raw type, as is currently done, the docs generate `addBootstrapListener(listener: function, once: boolean): void`, but later on in the parameters output it gives you this type data as well: `(event: SDKMessagePayload): void`.

Let's discuss if we want to open the door to these types which don't actually expose their internal structure in the docs in favor of making the source code simpler.

Reply:
Ah, yeah, agree that you were between a rock and a hard place.  Having it reference a type that you can't expand upon is no-bueno and opening the door to all interfaces (especially the ones that are private/internal) also seems sub-optimal.  I think we're good with this for now.

Maybe there's a way we can either opt-in/opt-out of interfaces/types in the future.  Maybe we even have to modify a copy of the declaration file or AST to get typedoc to generate what we ultimately want.

But, fine to handle this in pass two.